cluster.c

```diff
@@ -726,6 +872,39 @@ void ShardingInfoInit(RedisRaftCtx *rr)
     ShardingInfoReset(rr);
 }
 
+void FillShardSlots(RedisRaftCtx *rr, ShardGroup *sg)
+{
+     char * str = RedisModule_Strdup(rr->config->slot_config);
+
+    sg->slot_ranges_num = 1;
+    char * pos = str;
+    while ((pos = strchr(pos+1, ','))) {
+        sg->slot_ranges_num++;
+    }
+    sg->slot_ranges = RedisModule_Calloc(sg->slot_ranges_num, sizeof(ShardGroupSlotRange));
+
+    char * token = strtok(str, ",");
+    for(int i = 0; i < sg->slot_ranges_num; i++) {
+        unsigned long val;
+        if ((pos = strchr(token, ':'))) {
+            *pos = '\0';
+            val = strtoul(token, NULL, 10);
```

Comment:
Do we assume this has already been validated?

Reply:
added a function to validate